ecc/enclave/enclave_stub.go

```diff
@@ -367,50 +368,65 @@ func (e *StubImpl) Invoke(args []byte, pk []byte, shimStub shim.ChaincodeStubInt
 	signaturePtr := C.malloc(SIGNATURE_SIZE)
 	defer C.free(signaturePtr)
 
+	// get signed proposal
+	signedProposal, err := shimStub.GetSignedProposal()
+	if err != nil {
+		return nil, fmt.Errorf("cannot get signed proposal")
+	}
+	signedProposalBytes, err := proto.Marshal(signedProposal)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get signed proposal bytes")
+	}
+	signedProposalPtr := C.CBytes(signedProposalBytes)
+	defer C.free(unsafe.Pointer(signedProposalPtr))
+
+	//ASSUME HERE b64ChaincodeRequestMessagePtr is not the protobuf, so let's buildit (rmeove block later)
+	crmProto := &fpcpb.ChaincodeRequestMessage{
+		EncryptedRequest: args,
+	}
+	crmProtoBytes, err := proto.Marshal(crmProto)
+	if err != nil {
+		return nil, fmt.Errorf("marshal error")
+	}
+	b64crm := base64.StdEncoding.EncodeToString(crmProtoBytes)
+	//C.free(unsafe.Pointer(b64ChaincodeRequestMessagePtr))
+	b64ChaincodeRequestMessagePtr = C.CString(b64crm)
+	//REMOVE BLOCK ABOVE ONCE ARGS is the protobuf
+
 	e.sem.Acquire(context.Background(), 1)
 	// invoke enclave
 	invoke_ret := C.sgxcc_invoke(e.eid,
-		argsPtr,
-		pkPtr,
+		(*C.uint8_t)(signedProposalPtr),
+		C.uint32_t(len(signedProposalBytes)),
+		b64ChaincodeRequestMessagePtr,
+		C.uint32_t(len(b64crm)),
 		(*C.uint8_t)(responsePtr), C.uint32_t(MAX_RESPONSE_SIZE), &responseLenOut,
-		(*C.ec256_signature_t)(signaturePtr),
 		ctx)
 	e.sem.Release(1)
-	// Note: we do try to return the response in all cases, even then there is an error ...
-	var sig []byte = nil
-	var err error
-	if invoke_ret == 0 {
-		sig, err = crypto.MarshalEnclaveSignature(C.GoBytes(signaturePtr, C.int(SIGNATURE_SIZE)))
-		if err != nil {
-			sig = nil
-		}
-	} else {
-		err = fmt.Errorf("Invoke failed. Reason: %d", int(invoke_ret))
-		// TODO: ideally we would also sign error messages but would
-		// require including the error into the signature itself
-		// which has involves a rathole of changes, so defer to the
-		// time which design & refactor everything to be end-to-end
-		// secure ...
-	}
-	return C.GoBytes(responsePtr, C.int(responseLenOut)), sig, err
+	if invoke_ret != 0 {
+		return nil, fmt.Errorf("Invoke failed. Reason: %d", int(invoke_ret))
+	}
+	responseBytes = C.GoBytes(responsePtr, C.int(responseLenOut))
+	//return C.GoBytes(responsePtr, C.int(responseLenOut)), sig, err
+
+	//ASSUME HERE we get the b64 encoded response protobuf, pull encrypted response out and return it
+	chaincodeResponseMessageProtoBytes, err := base64.StdEncoding.DecodeString(C.GoStringN((*C.char)(responsePtr), C.int(responseLenOut)))
+	if err != nil {
+		return nil, fmt.Errorf("cannot decode b64 respose")
+	}
+	chaincodeResponseMessageProto := &fpcpb.ChaincodeResponseMessage{}
+	err = proto.Unmarshal(chaincodeResponseMessageProtoBytes, chaincodeResponseMessageProto)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal error")
+	}
+	responseBytes = chaincodeResponseMessageProto.EncryptedResponse
```

Comment:
```suggestion
        // TODO: this should be eventually be an (encrypted) fabric Response object rather than the response string ...
	responseBytes = chaincodeResponseMessageProto.EncryptedResponse
```

Reply:
done